### PR TITLE
refactor(linux): update variable configuration

### PR DESCRIPTION
**Pull request by jbrockopp**
Part of the effort for https://github.com/go-vela/community/issues/54

Dependent on https://github.com/go-vela/pkg-executor/pull/95

## Part 1

Removing the "shorthand" variable declarations in the `linux` executor.

Throughout the codebase, the first logic of each function is creating short variable declarations:

https://github.com/go-vela/pkg-executor/blob/384166ec013a3d18b61b5a4bab43a13dbb32c705/executor/linux/build.go#L24-L27

The problem with this (beyond the extra LOC and variables), is then we are left to update the client with that short variable:

https://github.com/go-vela/pkg-executor/blob/384166ec013a3d18b61b5a4bab43a13dbb32c705/executor/linux/build.go#L50

This attempts to make the variable stored in the client the "source of truth" so to speak since we always reference that.

## Part 2

Introducing a consistent `_<resource>` pattern for the `local` executor variables in the code:

* `_init`
* `_log`
* `_secret`
* `_service`
* `_stage`
* `_step`

### CODE REVIEW

File: executor/linux/build.go [hunk 1]
```diff
@@ -21,162 +21,140 @@ import (
...
 
-	// add the init container to secrets client
-	c.init = init
-
-	return nil
+	return c.err
 }
 
 // PlanBuild prepares the build for execution.
 func (c *client) PlanBuild(ctx context.Context) error {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply executor/linux/build.go:67]: Fixed 👍

File: executor/linux/secret.go [hunk 2]
```diff
@@ -421,18 +392,18 @@ func (s *secretSvc) stream(ctx context.Context, ctn *pipeline.Container) error {
 // TODO: Evaluate pulling this into a "bool" types function for injecting
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply executor/linux/secret.go:393]: Fixed 👍

File: executor/linux/secret.go [hunk 3]
```diff
@@ -398,13 +369,13 @@ func (s *secretSvc) stream(ctx context.Context, ctn *pipeline.Container) error {
...
 			// https://pkg.go.dev/github.com/go-vela/types/library?tab=doc#Log.AppendData
-			l.AppendData(logs.Bytes())
+			_log.AppendData(logs.Bytes())
 
 			logger.Debug("appending logs")
 			// send API call to append the logs for the init step
 			//
 			// https://pkg.go.dev/github.com/go-vela/sdk-go/vela?tab=doc#LogService.UpdateStep
-			l, _, err = s.client.Vela.Log.UpdateStep(r.GetOrg(), r.GetName(), b.GetNumber(), s.client.init.Number, l)
+			_log, _, err = s.client.Vela.Log.UpdateStep(s.client.repo.GetOrg(), s.client.repo.GetName(), s.client.build.GetNumber(), s.client.init.Number, _log)
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply executor/linux/secret.go:380]: Fixed 👍

File: executor/linux/secret.go [hunk 4]
```diff
@@ -274,7 +248,7 @@ func (s *secretSvc) exec(ctx context.Context, p *pipeline.SecretSlice) error {
 
 // pull defines a function that pulls the secrets from the server for a given pipeline.
 func (s *secretSvc) pull(secret *pipeline.Secret) (*library.Secret, error) {
-	sec := new(library.Secret)
+	_secret := new(library.Secret)
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply executor/linux/secret.go:252]: Fixed 👍

File: executor/linux/secret.go [hunk 5]
```diff
@@ -65,37 +63,11 @@ func (s *secretSvc) create(ctx context.Context, ctn *pipeline.Container) error {
 		return err
 	}
 
-	logger.Debug("marshaling configuration")
-	// marshal container configuration
-	body, err := json.Marshal(ctn)
+	logger.Debug("substituting container configuration")
```
Comment (matt-fevold) [inline executor/linux/secret.go:66]: I'm not really sure what's going on with the changes to this file. 
Unrelated - also looks like on line 48 there is a ref to a hard coded vela version.

Comment (jbrockopp) [reply executor/linux/secret.go:66]: > I'm not really sure what's going on with the changes from json.Marshal(ctn) -> ctn.substitute()

Yeah that's my fault 😞 

That is a continuation of https://github.com/go-vela/pkg-executor/pull/86 which I missed in the original PR.

More information can be found here:

https://github.com/go-vela/types/pull/129

But tl;dr is that we moved this chunk of logic into [go-vela/types](https://github.com/go-vela/types)

> Unrelated to that - also looks like on line 48 there is a ref to a hard coded vela version. Should that be changed too?

Yeah at some point that has to change but I haven't come up with a way to refactor that yet.

Hopefully in time, I'll have an idea on how we can fix that 😅

**Review by matt-fevold — APPROVED** [review]

File: executor/linux/secret.go (hunk 5 above)
Comment (matt-fevold) [reply executor/linux/secret.go:66]: thanks for the info!

**Review by wass3r — APPROVED** [review]

**Pull request merged**